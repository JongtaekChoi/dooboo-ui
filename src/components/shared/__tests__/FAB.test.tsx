import * as React from 'react';
import Fab, { FABProps } from '../FAB';

import { IC_EDIT } from '../Icons';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

let props: FABProps;
let component: React.ReactElement;
// let testingLib: RenderResult;

describe('[Fab] render', () => {
  beforeEach(() => {
    props = {
      src: IC_EDIT,
    };
    component = <View style={{ flex: 1 }}><Fab {...props} /></View>;
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON | null = renderer
      .create(component)
      .toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  // describe('interactions', () => {
  //   beforeEach(() => {
  //     testingLib = render(component);
  //   });

  //   it('should simulate onClick', () => {
  //     const btn = testingLib.queryByTestId('btn');
  //     act(() => {
  //       fireEvent.press(btn);
  //     });
  //     expect(cnt).toBe(3);
  //   });
  // });
});
