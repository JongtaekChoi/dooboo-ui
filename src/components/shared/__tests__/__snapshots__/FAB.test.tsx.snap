// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Fab] render renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 15,
          "position": "absolute",
          "right": 15,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 50,
          },
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "opacity": 1,
            "shadowRadius": 1,
            "width": 50,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../icons/ic_edit.png",
            }
          }
          style={
            Object {
              "borderRadius": 25,
              "height": 50,
              "shadowRadius": 1,
              "width": 50,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
